the-collab-lab/tcl-22-smart-shopping-list
# prevent duplicate entries in shopping list

## Description


Add logic to submit handler to check the user's entry against the database and prevent duplicate entries (including cases where capitalization and punctuation are different)

## Related Issue


Closes #7 

## Acceptance Criteria



- Show an error message if the user tries to submit a new item that has the exact same name as an existing item
- Show an error message if the user tries to submit a new item that has the same name as an existing item, where capitalization has been normalized and punctuation has been removed
- The user’s original input should be what gets saved in the database

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓  | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before


![Screen Cast 2021-04-20 at 7 32 23 PM](https://user-images.githubusercontent.com/58532369/115476195-5e958b00-a20f-11eb-91fd-a6445d75ccef.gif)


### After


![Screen Cast 2021-04-20 at 7 57 47 PM](https://user-images.githubusercontent.com/58532369/115477840-d1eccc00-a212-11eb-82d7-9b659fd1bbc6.gif)


## Testing Steps / QA Criteria



To test the feature, choose one of the following methods:

1. Follow the link to see the preview: https://tcl-22-shopping-list--pr27-dt-cf-prevent-duplic-4qgl7g7w.web.app
2. Launch the app locally from your terminal:
- Pull down this branch with git pull origin dt-cf-prevent-duplicate and check that branch out with git checkout origin dt-cf-prevent-duplicate
- Run `npm install` to update dependencies
- Run `npm start` to launch the app in your browser at http://localhost:3000
- When on application, click on 'Add Groceries' link in navbar to see user form. Fill user form with something that is already in 'Grocery List' and click 'Submit.' You should receive an error message.
